uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/performance/DeviceMonitor.java

```diff
@@ -0,0 +1,219 @@
+import java.util.regex.Pattern;
+
+/**
+ * Collects data for the current process for different {@link PerformanceProfile}. Additional profiles can included and
+ * their implementation done in {@link PerformanceProfileMonitor} then passed through the method {@link DeviceMonitor#setProfiler(PerformanceProfile, PerformanceProfileMonitor)}
+ */
+public class DeviceMonitor {
+
+    private static final String packageName = ApplicationProvider.getApplicationContext().getPackageName();
+    private static final HashMap<PerformanceProfile, PerformanceProfileMonitor<?>> profilers = new HashMap<>();
+    private static ProcessInfo processInfo = null;
+
+    static {
+        loadProcessInfo();
+        setProfiler(PerformanceProfile.CPU, new CPUMonitor());
+        setProfiler(PerformanceProfile.MEMORY, new MemoryMonitor());
+        setProfiler(PerformanceProfile.NETWORK, new NetworkUsageMonitor());
+    }
+
+    /**
+     * Get the device name string
+     *
+     * @return
+     */
+    public static String getDeviceName() {
+        String manufacturer = Build.MANUFACTURER;
+        String model = Build.MODEL;
+        if (model.toLowerCase().startsWith(manufacturer.toLowerCase())) {
+            return capitalize(model);
+        } else {
+            return capitalize(manufacturer) + " " + model;
+        }
+    }
+
+
+    private static String capitalize(String s) {
+        if (s == null || s.length() == 0) {
+            return "";
+        }
+        char first = s.charAt(0);
+        if (Character.isUpperCase(first)) {
+            return s;
+        } else {
+            return Character.toUpperCase(first) + s.substring(1);
+        }
+    }
+
+    /**
+     * Gets the current application's uid
+     *
+     * @return the application uid
+     */
+    public static int getApplicationUid() {
+        final Pattern pattern = Pattern.compile("userId=(\\d+)");
+        final String command = "dumpsys package " + packageName;
+
+        final Matcher matcher = pattern.matcher(AdbShellUtils.executeShellCommand(command));
+
+        if (matcher.find()) {
+            return Integer.parseInt(Objects.requireNonNull(matcher.group(1)));
+        }
+        return 0;
+    }
+
+    /**
+     * Get the total System memory
+     *
+     * @return
+     */
+    public static Long getTotalMemory() {
+        return processInfo.getTotalSystemMemory();
+    }
+
+    /**
+     * Get the current CPU usage
+     *
+     * @return the current cpu usage as a percentage e.g. 3.3%
+     */
+    public static Double getCpuUsage() {
+        return getStats(PerformanceProfile.CPU, Double.class);
+    }
+
+    /**
+     * Get the current memory usage
+     *
+     * @return the current memory usage in KB
+     */
+    public static Long getMemoryUsage() {
+        return getStats(PerformanceProfile.MEMORY, Long.class);
+    }
+
+    /**
+     * Get the current network info
+     *
+     * @return the information regarding the network usage
```

Comment:
Please add an `{@link <DataType>}` to reference to the return type in the javadoc here and everywhere else as well